hw10/hw10_t01.py

```diff
@@ -0,0 +1,218 @@
+
+            current_value = float(
+                soup.find(
+                    "span", attrs={"class": "price-section__current-value"}
+                ).string.replace(",", "")
+            )
+
+            code = list(
+                soup.find("span", attrs={"class": "price-section__category"}).strings
+            )[-2].strip(", ")
+
+            p_e = get_parameter_from_company_page(soup, "P/E Ratio")
+
+            year_low = get_parameter_from_company_page(soup, "52 Week Low")
+            year_high = get_parameter_from_company_page(soup, "52 Week High")
+            if all((year_low, year_high)):
+                potential_profit = round((year_high - year_low) * usd, 2)
+            else:
+                potential_profit = None
+            company_info = {
+                "name": name,
+                "code": code,
+                "price": current_value,
+                "P/E": p_e,
+                "growth": growth,
+                "potential profit": potential_profit,
+            }
+            evaluate_top10(company_info)
+
+
+def get_parameter_from_company_page(soup: BeautifulSoup, string: str) -> float:
+    try:
+        return float(
+            list(
+                soup.find(
+                    "div",
+                    attrs={"class": "snapshot__header"},
+                    string=string,
+                ).parent.stripped_strings
+            )[0].replace(",", "")
+        )
+    except AttributeError:
+        pass
+
+
+def evaluate_top10(company: dict) -> None:
+    evaluate_top10_price(company)
+    evaluate_top10_pe(company)
+    evaluate_top10_growth(company)
+    evaluate_top10_potential_profit(company)
+
+
+def evaluate_top10_price(company: dict) -> None:
+    for i in range(10):
+        if company["price"] and company["price"] > highest_price[i]["price"]:
+            highest_price.insert(i, company)
+            highest_price.pop()
+            break
+
+
+def evaluate_top10_pe(company: dict) -> None:
+    for i in range(10):
+        if company["P/E"] and company["P/E"] < lowest_pe[i]["P/E"]:
+            lowest_pe.insert(i, company)
+            lowest_pe.pop()
+            break
+
+
+def evaluate_top10_growth(company: dict) -> None:
+    for i in range(10):
+        if company["growth"] and company["growth"] > highest_growth[i]["growth"]:
+            highest_growth.insert(i, company)
+            highest_growth.pop()
+            break
+
+
+def evaluate_top10_potential_profit(company: dict) -> None:
+    for i in range(10):
+        if (
+            company["potential profit"]
+            and company["potential profit"]
+            > highest_potential_profit[i]["potential profit"]
+        ):
+            highest_potential_profit.insert(i, company)
+            highest_potential_profit.pop()
+            break
+
+
+def write_into_files() -> None:
+    with open("highest_price.json", "w") as file:
+        file.write(json.dumps(highest_price))
```

Comment:
You can do it in a more simple way. Please check the JSON module documentation.

Reply:
> You can use tuples (-company["price"], company['code'], company)
Tuples are compared from left to right. You can use +- to change min/max, the second element is guaranteed to be unique by task definition so we will never compare third one.
>
I don't understand this idea yet. So I should have 4 tuples instead of 4 lists to form a JSON or single company info should be a tuple?

Reply:
You have left comments in a strange place.  Originally it was the response on how you could use heap.

Reply:
No reply window for that comment(

Reply:
> No reply window for that comment(

So you decided to put a replay to a random comment?  It is possible to add general comments, scroll down the page.